how-to-docs/cert-manager.md

```diff
@@ -0,0 +1,235 @@
+
+```YAML
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: tls
+spec:
+  secretName: tls
+  issuerRef:
+    name: platform-tls
+    kind: ClusterIssuer
+  commonName: site.svc.project.cluster.local
+  dnsNames:
+  - localhost
+  - 127.0.0.1
+```
+
+This would create a kubernetes secret named `tls` in your namespace with the signed certificate. An interesting thing to note here although this is using the ClusterIssuer platform-ca created by the ACP team, there is nothing stopping a project from creating a local Issuer for the own project. So for example.
+
+```YAML
+---
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: project-ca
+spec:
+  ca:
+    secretName: project-ca
+---
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: tls
+spec:
+  secretName: tls
+  issuerRef:
+    name: project-ca
+    # @Note: we have change from ClusterIssuer to a local Issuer
+    kind: Issuer
+  commonName: site.svc.project.cluster.local
+  dnsNames:
+  - localhost
+  - 127.0.0.1
+```
+
+#### **As a developer I want to retrieve a certificate for my external service**
+
+Lets assume we have an externally facing site which we wish to expose via ingress and we want a valid Letsencrypt certificate. Two have two things to note here;
+
+- the enable annotation `certmanager.k8s.io/enabled` which is a toggle to ask cert-manager to handle this ingress resource.
+- the Letsencrypt challenge `certmanager.k8s.io/acme-challenge-type`; note, technically the challenge is not required as `http01` is the default type, but I'm adding to highlight.
+
+**When to use a HTTP challenge**
+
+Assuming the site is externally facing i.e. the ingress class on the ingress is `kubernetes.io/ingress.class: ingress-external` you should always default to using a http01 challenge. Things change, however, when the site is internal / behind the vpn. In order to handle the challenge when behind the VPN you need to switch to using a DNS challenge. With ingress this is done by adding the annotation `certmanager.k8s.io/acme-challenge-type: dns`. **Very Important** this assumes you have contacted the ACP beforehand and we have either added the domain to our route53 or added the dns provider as a challenge provider.
+
+```YAML
+apiVersion: extensions/v1beta1
+  kind: Ingress
+  metadata:
+    annotations:
+      # @NOTE: we choose http challenge
+      certmanager.k8s.io/acme-challenge-type: http01
+      # @NOTE: this will enable the cert-manager to handle this resource
+      certmanager.k8s.io/enabled: "true"
+      ingress.kubernetes.io/affinity: cookie
+      ingress.kubernetes.io/force-ssl-redirect: "true"
+      ingress.kubernetes.io/secure-backends: "true"
```

Comment:
remove secure-backends and replace with `ingress.kubernetes.io/backend-protocol: "HTTPS"`